src/tests/fuzzHttp1Parser.cc

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) 1996-2023 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#include "squid.h"
+
+#include "compat/cppunit.h"
+#include "debug/Stream.h"
+#include "http/one/RequestParser.h"
+#include "http/RequestMethod.h"
+#include "MemBuf.h"
+#include "SquidConfig.h"
+
+#define kMinInputLength 10
+#define kMaxInputLength 5120
+
+static bool setup_done = false;
+
+extern "C" int LLVMFuzzerTestOneInput(const char *data, size_t size)
+{
+
+    if (size < kMinInputLength || size > kMaxInputLength) {
+        return 0;
+    }
+
+    if (!setup_done){
+        Mem::Init();
+        setup_done = true;
+
+        // default to strict parser. set for loose parsing specifically where behaviour differs.
+        Config.onoff.relaxed_header_parser = 0;
+
+        Config.maxRequestHeaderSize = 1024; // XXX: unit test the RequestParser handling of this limit
+    }
+
+    SBuf ioBuf;
+    Http1::RequestParser hp;
+
+    ioBuf.clear();
+    ioBuf.append(data, size);
+
+    hp.clear();
+    hp.parse(ioBuf);
```

Comment:
If fuzzer is able to find a Squid bug, how will that bug manifest itself in this test? A crash?

What happens if Http::One::RequestParser::parse() throws an exception?

BTW, did the proposed tests flagged any problems in covered Squid parsers?

Reply:
> If fuzzer is able to find a Squid bug, how will that bug manifest itself in this test? A crash?

A crash report is generated by `oss-fuzz infra` and an email is sent to the [contact](https://google.github.io/oss-fuzz/getting-started/new-project-guide/#primary). After downloading the crash file, the bug can be [reproduced](https://google.github.io/oss-fuzz/advanced-topics/reproducing).




> What happens if Http::One::RequestParser::parse() throws an exception?

Done thanks for pointing it out  https://github.com/squid-cache/squid/pull/1531/commits/2aec1c227fbe198d9e040bb20ac4ba05aa640c9c

> BTW, did the proposed tests flagged any problems in covered Squid parsers?

Yes, a heap-buffer-overflow in `rfc1035MessageUnpack`;  I send a bug report to `squid-bugs@lists.squid-cache.org`